src/api-client/withApollo.tsx

```diff
@@ -67,11 +68,12 @@ export function getApolloClient() {
   return _apolloClient;
 }
 
-export const withApollo = (Comp: NextPage) =>
-  function ApolloWrapper(props: object) {
-    return (
-      <ApolloProvider client={getApolloClient()}>
-        <Comp {...props} />
-      </ApolloProvider>
-    );
-  };
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function withApollo<T extends NextPage<any>>(Component: T): T {
```

Comment:
Fun with generics so uses of this HOC carry forward props from the wrapped component

Reply:
Is this not best achieved with the children props?

Reply:
sorry I'm not following, could I get some more detail?

Reply:
[React composition works by passing the special `children` prop](https://react.dev/learn/passing-props-to-a-component). Here I wonder why withAppolo is not following such a pattern:

```tsx
function WithAppolo({ children }: PropsWithChildren<{}>){
  return <Provider>
    {children}
  </Provider>
}
```